### PR TITLE
recommend better application name based on the component url

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Fixes 

https://issues.redhat.com/browse/RHTAPBUGS-237

## Description

Use repo name from component url as a default application name, and allow user to change it in the review step, if it is desired.

## Type of change

- [x] Bugfix

## Screen shots / Gifs for design review 

https://user-images.githubusercontent.com/9964343/237036607-20b76e49-c5a9-43c3-9253-25f10ebfe211.mov

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
cc: @rohitkrai03 @christianvogt

**Review by jeff-phillips-18 — APPROVED** [review]
/lgtm

File: src/components/ImportForm/SourceSection/SourceSection.tsx [hunk 1]
```diff
@@ -91,7 +94,9 @@ const SourceSection: React.FC<SourceSectionProps> = () => {
     if (accessCheckLoaded) {
       if (isRepoAccessible) {
         try {
-          const { organization } = gitUrlParse(sourceUrl);
+          const { organization, name } = gitUrlParse(sourceUrl);
+          !inAppContext && setFieldValue('application', incrementNameCount(applications, name));
```
Comment (rohitkrai03) [inline src/components/ImportForm/SourceSection/SourceSection.tsx:98]: Do we actually need to do this in SourceSection? Maybe just update the application name in `ApplicationSection`. The hook `useApplicationName` can take a Git URL and return back a valid application based on git repo.

File: src/components/ImportForm/utils/useValidApplicationName.ts [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
+  let name = prefferedName || BASE_NAME,
+    i = 1;
+  while (applications.some((app) => app.metadata.name === name)) {
+    name = `${prefferedName || BASE_NAME}-${i}`;
+    i++;
+  }
+  return name;
+};
+
 export const useValidApplicationName = (): [string, boolean, unknown] => {
```
Comment (rohitkrai03) [inline src/components/ImportForm/utils/useValidApplicationName.ts:21]: You would also need to remove the use of this hook from `ImportForm` where it is sending `recommendedApplicationName`.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: src/components/ImportForm/SourceSection/SourceSection.tsx (hunk 1 above)
Comment (rottencandy) [reply src/components/ImportForm/SourceSection/SourceSection.tsx:98]: I think we can also update it for sample import flow this way

Comment (karthikjeeyar) [reply src/components/ImportForm/SourceSection/SourceSection.tsx]: Moved the hook usage to review section. For samples flow, I am retaining the same recommendation logic as before.

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Pull request merged**